File: java/phonebook/src/com/phonebook/Timer.java
package com.phonebook;

// Faithful port of include/timer.hpp.
public final class Timer {
    private Timer() {}

    public static double timeIt(Runnable work) {
        long t0 = System.nanoTime();
        work.run();
        long t1 = System.nanoTime();
        return (t1 - t0) / 1e6;
    }

    public static double benchmark(Runnable work, int repeats) {
        if (repeats <= 0) {
            return 0.0;
        }
        double best = Double.MAX_VALUE;
        for (int r = 0; r < repeats; r++) {
            double t = timeIt(work);
            if (t < best) {
                best = t;
            }
        }
        return best;
    }

    public static double benchmark(Runnable work) {
        return benchmark(work, 5);
    }

    public static void printTaskDuration(Runnable work) {
        System.out.println("\nTook: " + timeIt(work) + "ms.");
    }
}

File: javascript/phonebook/src/timer.js
'use strict';

// Faithful port of include/timer.hpp.
const { performance } = require('node:perf_hooks');

function timeIt(work) {
  const t0 = performance.now();
  work();
  const t1 = performance.now();
  return t1 - t0;
}

function benchmark(work, repeats = 5) {
  if (repeats <= 0) return 0.0;
  let best = Number.POSITIVE_INFINITY;
  for (let r = 0; r < repeats; r++) {
    const t = timeIt(work);
    if (t < best) best = t;
  }
  return best;
}

function printTaskDuration(work) {
  console.log(`\nTook: ${timeIt(work)}ms.`);
}

module.exports = { timeIt, benchmark, printTaskDuration };
